dynamometer-workload/src/main/java/com/linkedin/dynamometer/workloadgenerator/CreateFileMapper.java

```diff
@@ -60,64 +57,41 @@ public String getDescription() {
 
   @Override
   public List<String> getConfigDescriptions() {
-    return Lists.newArrayList(
-        NUM_MAPPERS_KEY + " (required): Number of mappers to launch.",
-        DURATION_MIN_KEY + " (required): Number of minutes to induce workload for.",
-        SHOULD_DELETE_KEY + " (default: " + SHOULD_DELETE_DEFAULT + "): If true, delete the files after creating " +
-            "them. This can be useful for generating constant load without increasing the number of file objects.",
-        FILE_PARENT_PATH_KEY + " (default: " + FILE_PARENT_PATH_DEFAULT +
-            "): The root directory in which to create files."
-    );
+    List<String> baseList = TimedInputFormat.getConfigDescriptions();
+    baseList.add(SHOULD_DELETE_KEY + " (default: " + SHOULD_DELETE_DEFAULT +
+        "): If true, delete the files after creating them. This can be useful for generating constant load " +
+        "without increasing the number of file objects.");
+    baseList.add(FILE_PARENT_PATH_KEY + " (default: " + FILE_PARENT_PATH_DEFAULT +
+        "): The root directory in which to create files.");
+    return baseList;
   }
 
   @Override
   public boolean verifyConfigurations(Configuration conf) {
-    return conf.get(NUM_MAPPERS_KEY) != null && conf.get(DURATION_MIN_KEY) != null;
+    return TimedInputFormat.verifyConfigurations(conf);
   }
 
   @Override
-  public void map(NullWritable key, NullWritable value, Mapper.Context mapperContext)
-      throws IOException, InterruptedException {
-    taskID = mapperContext.getTaskAttemptID().getTaskID().getId();
-    conf = mapperContext.getConfiguration();
+  public void setup(Mapper.Context context) throws IOException {
+    Configuration conf = context.getConfiguration();
     String namenodeURI = conf.get(WorkloadDriver.NN_URI);
-    startTimestampMs = conf.getLong(WorkloadDriver.START_TIMESTAMP_MS, -1);
-    fileParentPath = conf.get(FILE_PARENT_PATH_KEY, FILE_PARENT_PATH_DEFAULT);
-    shouldDelete = conf.getBoolean(SHOULD_DELETE_KEY, SHOULD_DELETE_DEFAULT);
-    int durationMin = conf.getInt(DURATION_MIN_KEY, -1);
-    if (durationMin < 0) {
-      throw new IOException("Duration must be positive; got: " + durationMin);
-    }
-    endTimeStampMs = startTimestampMs + TimeUnit.MILLISECONDS.convert(durationMin, TimeUnit.MINUTES);
     fs = FileSystem.get(URI.create(namenodeURI), conf);
-    System.out.println("Start timestamp: " + startTimestampMs);
-
-    long currentEpoch = System.currentTimeMillis();
-    long delay = startTimestampMs - currentEpoch;
-    if (delay > 0) {
-      System.out.println("Sleeping for " + delay + " ms");
-      Thread.sleep(delay);
-    }
+    shouldDelete = conf.getBoolean(SHOULD_DELETE_KEY, SHOULD_DELETE_DEFAULT);
+    String fileParentPath = conf.get(FILE_PARENT_PATH_KEY, FILE_PARENT_PATH_DEFAULT);
+    int taskID = context.getTaskAttemptID().getTaskID().getId();
+    filePrefix = new Path(new Path(new Path(fileParentPath), "mapper" + taskID), "file");
```

Comment:
why not just `new Path(a + b + c)`? this new path(new path(new path))) looks weird

Reply:
Typically we use the `new Path(parent, child)` syntax to let the `Path` class handle missing/extra separators, avoiding explicitly writing out the separator char, etc. But I agree that it gets a bit much here. Updated.